app/src/main/java/com/github/se/cyrcle/model/parking/ParkingViewModel.kt

```diff
@@ -194,31 +189,25 @@ class ParkingViewModel(
         Point.fromLngLat(
             maxOf(startPos.longitude(), endPos.longitude()),
             maxOf(startPos.latitude(), endPos.latitude()))
-
     // Get all tiles that are in the rectangle
-    tilesToDisplay = Tile.getAllTilesInRectangle(bottomLeft, topRight)
-    // Used to keep track of when the last request has finished.
+    val tilesToDisplay = TileUtils.getAllTilesInRectangle(bottomLeft, topRight)
     var nbRequestLeft = tilesToDisplay.size
 
     tilesToDisplay.forEach { tile ->
 
-      // Get parkings from memory cache
-      if (tileCache.containsKey(tile.uid)) {
-        _rectParkings.value += tileCache[tile.uid]!!.parkings
+      //  Cache hit
+      if (tilesToParking.containsKey(tile)) {
+        _rectParkings.value += tilesToParking[tile]!!
         updateClosestParkings(--nbRequestLeft)
 
-        // Get parkings from repository
+        // Cache miss
```

Comment:
Just a quick note (non-blocking), is that an explanation for an `if-else` statement should be written as follow
```
if (condition) {
    // Explanation
    ... code ...
} else {
    // Explanation
    ... code ...
}
```
Instead of above the branching, as this would better indicate the whole thing
```
// Absolute value of x
if (x < 0) {
    x =-x;
} else {
    x = x;
}
```